arquin/device/main.py

```diff
@@ -0,0 +1,41 @@
+import networkx as nx
+import numpy as np
+
+from arquin.module.main import Module
+
+class Device:
+    def __init__(self, device_graph, module_graphs):
+        '''
+        device_graph: graph of the device. Each module is a node. Inter-module connections are the edges.
+        module_graphs: list of graphs of each module
+        '''
+        assert len(module_graphs) == device_graph.size()
+        self.modules = self._build_modules(module_graphs=module_graphs)
+        self.global_edges, self.local_edges = self._connect_modules(device_graph=device_graph)
+        self.abstract_global_edges = device_graph.edges
+        self.size = sum([module.size for module in self.modules])
+    
+    def _build_modules(self, module_graphs):
+        modules = []
+        offset = 0
+        for module_graph in module_graphs:
+            module = Module(graph=module_graph, offset=offset)
+            offset += module.size
+            modules.append(module)
+        return modules
+    
+    def _connect_modules(self, device_graph):
+        global_edges = []
+        for global_edge in device_graph.edges:
+            module_idx_l, module_idx_r = global_edge
+            module_l_qubit = np.random.choice(self.modules[module_idx_l].qubits) + self.modules[module_idx_l].offset
+            module_r_qubit = np.random.choice(self.modules[module_idx_r].qubits) + self.modules[module_idx_r].offset
+            global_edges.append([module_l_qubit, module_r_qubit])
```

Comment:
The way `module_l_qubit` and `module_r_qubit` are computed here does not follow the description given in the conversation tab:

> device.global_edges, device.local_edges and module.edges are all represented by physical qubits. Physical qubits are labelled module_i_j for qubit j in module i.

I think it's fine to use either definition because they contain the same information. We can write functions in `Device` that can convert between the two representations because they are likely better for different things.

I am currently merging my changes into this branch, so let's discuss after I've done that because I've also slightly changed how `Device` is instantiated (to avoid the random assignment of module qubits)

Reply:
In fact, I think that `_connect_modules` can be split into two functions that perform this conversion for us. 

The first function should convert "module format qubits" to "device format qubits". So it should take in tuples (module_i, qubit_j of module_i) and return an integer corresponding to the physical device qubit label.

The second function should construct the "physical qubit graph" for the device.